### PR TITLE
feat: fixed name cloning to target the task name

### LINKED ISSUE #883: Tasks: Cloned task should have a different name than the original
There is an option to clone a task in the UI, but the name of the new task is the same as the old task, making it hard to distinguish.

other resources, when cloned, give the clone a "Clone 1" at the end. We should also do that with tasks.

### PR DESCRIPTION
**Pull request by Asalem1**
Closes #883 

This time it should close the issue for real. Updated the regex to properly assert replacing the task name declaration rather than the first string like it was in the previous PR (#1063)

### CODE REVIEW

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -227,13 +227,15 @@ export const createTask = (
...
   name: string = '🦄ask'
 ): Cypress.Chainable<Cypress.Response> => {
-  const flux = `option task = {
-    name: "${name}",
-    every: 24h,
-    offset: 20m
-  }
-  from(bucket: "defbuck")
-        |> range(start: -2m)`
+  const flux = `import "csv"
```
Comment (Asalem1) [inline cypress/support/commands.ts:230]: Added a generic import here to complicate the task a bit and ensure that another string is declared prior to the task variable

File: src/tasks/actions/thunks.ts [hunk 2]
```diff
@@ -244,15 +245,36 @@ export const deleteTask = (taskID: string) => async (
...
-    const newTask = await api.postTask({data: resp.data})
+    const taskName = resp.data.name
+    const tasks = getAll<Task>(state, ResourceType.Tasks)
+    const allTaskNames = tasks.map(d => d.name)
+    const clonedName = incrementCloneName(allTaskNames, taskName)
+    const flux = resp.data.flux
+
+    // Searches for the beginning option task declaration, ignoring whitespace prior to the task name
+    const revisedFlux = flux.replace(
+      /option task\s*=\s*{\s*name:\s*".*?"/,
```
Comment (Asalem1) [inline src/tasks/actions/thunks.ts:268]: updated the regex to assert a more direct match for a task option rather

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 39 | :red_circle:
accessibility | 96 | :green_circle:
best-practices | 93 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :question:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1110)

**Comment by drdelambre** [comment]
i'm not following why we need to use regex for this.
we have all of the meta data required to define that option, and it's autogenerated from that metadata in the first place (as in.. it's not code that a user would particularly have interest in personalizing), so why dont we parse the whole task script into an AST, blow away that option's tree, and then rebuild the option definition and push it back into the AST?

that'd offload the parsing complexity / rules onto the flux parsing wasm and allow the flux team to control that complexity

I'm getting flashbacks from variable parsing issues that look similar to this

File: src/tasks/actions/thunks.ts [hunk 3]
```diff
@@ -244,15 +247,55 @@ export const deleteTask = (taskID: string) => async (
...
+    const {flux} = resp.data
+
+    const ast = parse(flux)
+
+    ast.body = ast.body.map(statement => {
+      if (
+        statement.type === 'OptionStatement' &&
+        statement.assignment.init &&
+        Array.isArray(statement.assignment.init.properties)
+      ) {
```
Comment (Asalem1) [inline src/tasks/actions/thunks.ts:275]: didn't abstract these because loops + functions !== "i'm melting"

File: src/tasks/actions/thunks.ts [hunk 4]
```diff
@@ -244,15 +247,55 @@ export const deleteTask = (taskID: string) => async (
...
+    const tasks = getAll<Task>(state, ResourceType.Tasks)
+    const allTaskNames = tasks.map(d => d.name)
+    const clonedName = incrementCloneName(allTaskNames, taskName)
+    const {flux} = resp.data
+
+    const ast = parse(flux)
+
+    ast.body = ast.body.map(statement => {
+      if (
+        statement.type === 'OptionStatement' &&
```
Comment (drdelambre) [inline src/tasks/actions/thunks.ts:272]: i think you also have to look for the option to be named `task` somewhere in here, or else it'll overwrite the variable `name` if:
```
option v = {
    name: 'yolo'
}

option task = {
    name: 'task to clone'
}
````

Comment (Asalem1) [reply src/tasks/actions/thunks.ts:272]: I think that's illegal in some states but better put up some guards

**Review by drdelambre — APPROVED** [review]

**Pull request merged**